pkg/controller/externalippool/controller.go

```diff
@@ -231,9 +232,13 @@ func (c *ExternalIPPoolController) createOrUpdateIPAllocator(ipPool *antreacrds.
 				// Don't use the IPv4 network's broadcast address.
 				var reservedIPs []net.IP
 				if utilnet.IsIPv4CIDR(ipNet) {
-					reservedIPs = append(reservedIPs, iputil.GetLocalBroadcastIP(ipNet))
+					lastAddr := make(net.IP, len(ipNet.IP.To4()))
+					if includeBorderIPs == false || (lastAddr != nil && (lastAddr[3] == 0 || lastAddr[3] == 255)) {
+						reservedIPs = append(reservedIPs, iputil.GetLocalBroadcastIP(ipNet))
+					}
 				}
-				return ipallocator.NewCIDRAllocator(ipNet, reservedIPs)
+				newIPAllocator, _ := ipallocator.NewCIDRAllocator(ipNet, reservedIPs)
+				return ipallocator.ValidateAllocatorforBorderIPs(newIPAllocator, ipNet, includeBorderIPs)
```

Comment:
Why doesn't it pass the parameter to `NewCIDRAllocator` but uses another function to exclude the IPs? It seems overcomplicated.

Reply:
because this function is called inside other functions also so changing its signature will break the code.